lib/fixed-points.nix

```diff
@@ -311,63 +316,137 @@ rec {
     );
 
   /**
-    Compose two extending functions of the type expected by 'extends'
-    into one where changes made in the first are available in the
-    'super' of the second
+    Compose two overlay functions and return a single overlay function that combines them.
+    For more details see: [composeManyExtensions](#function-library-lib.fixedPoints.composeManyExtensions).
+  */
+  composeExtensions =
+    f: g: final: prev:
+    let
+      fApplied = f final prev;
+      prev' = prev // fApplied;
+    in
+    fApplied // g final prev';
+
+  /**
+    Composes a list of [`overlays`](#chap-overlays) and returns a single overlay function that combines them.
 
+    :::{.note}
+    The result is produced by using the update operator `//`.
+    This means nested values of previous overlays are not merged recursively.
+    In other words, previously defined attributes are replaced, ignoring the previous value, unless referenced by the overlay; for example `final: prev: { foo = final.foo + 1; }`.
+    :::
 
     # Inputs
 
-    `f`
-
-    : 1\. Function argument
+    `extensions`
 
-    `g`
+    : A list of overlay functions
+      :::{.note}
+      The order of the overlays in the list is important.
+      :::
 
-    : 2\. Function argument
+    : Each overlay function takes two arguments, by convention `final` and `prev`, and returns an attribute set.
+      - `final` is the result of the fixed-point function, with all overlays applied.
+      - `prev` is the result of the previous overlay function.
 
-    `final`
 
-    : 3\. Function argument
+    # Type
 
-    `prev`
+    ```
+    # Pseudo code
+    let
+      #               final      prev
+      #                 ↓          ↓
+      OverlayFn :: ( { ... } -> { ... } -> { ... } );
+    in
+      composeManyExtensions :: [ OverlayFn ] -> OverlayFn
+    ```
 
-    : 4\. Function argument
-  */
-  composeExtensions =
-    f: g: final: prev:
-      let fApplied = f final prev;
-          prev' = prev // fApplied;
-      in fApplied // g final prev';
+    # Examples
+    :::{.example}
+    ## `lib.fixedPoints.composeManyExtensions` educational example without `lib.extends`
 
-  /**
-    Compose several extending functions of the type expected by 'extends' into
-    one where changes made in preceding functions are made available to
-    subsequent ones.
+    :::{.warning}
+    This is a purely educational example
```

Comment:
I suggest not relying on the current order of examples. Given examples are collected, they may be viewed out of context. Use a link instead of a verbal reference.